fuel-core/src/database/vm_database.rs

```diff
@@ -285,28 +256,25 @@ impl InterpreterStorage for VmDatabase {
 
         let mut current_key = U256::from_big_endian(start_key.as_ref());
 
-        let transaction = self.database.transaction();
```

Comment:
We use `VMDatabase` only with `DatabaseTransaction` right now. Also, the chance that we will use it directly with a storage database in the future is low.

Reply:
Makes sense, it's basically already batched most of the time even if not explicitly handled here.